src/main/java/com/hierynomus/smbj/utils/SmbFiles.java

```diff
@@ -64,4 +74,31 @@ public static int copy(File source, DiskShare share, String destPath, boolean ov
         return r;
     }
 
+    /**
+     * Create a set of nested sub-directories in the given path, for example, 2345 \ 3456 \ 4453 \ 123123.txt
+     */
+    public void mkdirs(DiskShare diskShare, String path) throws SMBApiException {
+        String currPath = path;
+        StringBuilder currDir = new StringBuilder();
+        while (path.startsWith("\\"))
+            currPath = currPath.substring(1, currPath.length() - 1 );
```

Comment:
Use `substring(int beginIndex)`, now you also chop off a character at the end of the string. This is why tests are important.

Reply:
fixed